File: Shinezone-U3D_dev_MOBA/myMOBA-4/Tools/Photon-OnPremise-Server-SDK_v4-0-29-11263/src-server/Plugins/CppCliPlugin/PluginFactory.cpp
#include "stdafx.h"
#include "PluginFactory.h"
#include "CppCliPlugin.h"

namespace CppCliPlugin
{

PluginFactory::PluginFactory()
{
}

IGamePlugin^ PluginFactory::Create(IPluginHost^ host, String^ pluginName, Collections::Generic::Dictionary<String^, String^>^ config, String^% errorMsg)
{
	CppCliPlugin^ res = gcnew CppCliPlugin();

	errorMsg = String::Empty;

	if (res != nullptr)
	{
		if (!res->SetupInstance(host, config, errorMsg))
		{
			return nullptr;
		}
	}
	return res;
}

}

File: Shinezone-U3D_dev_MOBA/myMOBA-4/Tools/Photon-OnPremise-Server-SDK_v4-0-29-11263/src-server/Plugins/WebHooks1.2/PluginFactory.cs
using System.Collections.Generic;

namespace Photon.Hive.Plugin.WebHooks
{
    public class PluginFactory : IPluginFactory
    {
        public IGamePlugin Create(IPluginHost gameHost, string pluginName, Dictionary<string, string> config, out string errorMsg)
        {
            var plugin = new WebHooksPlugin();
            if (plugin.SetupInstance(gameHost, config, out errorMsg))
            {
                return plugin;
            }
            return null;
        }
    }
}
